### ISSUE TITLE
Missing space after newlines in JSX literals

### ISSUE THREAD

**Issue by caseywebdev** [6to5/6to5#668]
Given this JSX

```jsx
<div>
  foo
  bar
</div>
```

The rendered html should look like `foo bar` on a page.

[Facebook adds a space between foo and bar](http://facebook.github.io/react/jsx-compiler.html)

```js
React.createElement("div", null, 
	"foo" + ' ' +
    "bar"
)
```

[But 6to5 doesn't](http://6to5.org/repl/#?experimental=true&playground=true&evaluate=true&loose=false&code=%3Cdiv%3E%0A%20%20foo%0A%20%20bar%0A%3C%2Fdiv%3E)

```js
"use strict";

React.createElement(
  "div",
  null,
  "foo",
  "bar"
);
```

The latter results in inappropriately spaced html `foobar`.

This can be worked around by keeping text one one line (which can get very long) or manually inserting `{' '}` after each newline (which is very tedious and ugly).

**Comment by sebmck** [comment]
The whitespacing rules of JSX aren't documented very well. I'll have to revaluate the current whitespacing rules as 99% of them are copy-paste from [jstransform](https://github.com/facebook/react/blob/master/vendor/fbtransform/transforms/xjs.js) but apparently that might not be enough.

**Comment by caseywebdev** [comment]
[Here is the bit](https://github.com/facebook/react/blob/63c3461af7a2e8f0a8d4c7399db5e82e4a98ce4a/vendor/fbtransform/transforms/xjs.js#L52) that seems to be missing in 6to5.

**Comment by sebmck** [comment]
I've added the relevant behaviour but instead of adding an ` + " "` it just appends a space onto the literal. I'm really not sure why this isn't done in the official JSX transformer as it's much nicer.

**Comment by caseywebdev** [comment]
I agree that is nicer. As always thanks for the quick fix!

**Comment by sebmck** [comment]
Fixed as of 3.3.5, thanks!
